crates/core/component/shielded-pool/src/component/rpc/transfer_query.rs

```diff
@@ -0,0 +1,91 @@
+use anyhow::Context;
+use async_trait::async_trait;
+use futures::StreamExt;
+use ibc_proto::ibc::applications::transfer::v1::query_server::Query as TransferQuery;
+use ibc_proto::ibc::apps::transfer::v1::{
+    DenomTrace, QueryDenomHashRequest, QueryDenomHashResponse, QueryDenomTraceRequest,
+    QueryDenomTraceResponse, QueryDenomTracesRequest, QueryDenomTracesResponse,
+    QueryEscrowAddressRequest, QueryEscrowAddressResponse, QueryParamsRequest, QueryParamsResponse,
+    QueryTotalEscrowForDenomRequest, QueryTotalEscrowForDenomResponse,
+};
+use penumbra_asset::asset::Metadata;
+use penumbra_proto::StateReadProto as _;
+
+use crate::state_key;
+
+use super::Server;
+
+// use super::IbcQuery;
+
+#[async_trait]
+// impl<HI: HostInterface + Send + Sync + 'static> TransferQuery for Server {
+impl TransferQuery for Server {
+    async fn total_escrow_for_denom(
+        &self,
+        _: tonic::Request<QueryTotalEscrowForDenomRequest>,
+    ) -> std::result::Result<tonic::Response<QueryTotalEscrowForDenomResponse>, tonic::Status> {
+        unimplemented!()
+    }
+
+    async fn escrow_address(
+        &self,
+        _: tonic::Request<QueryEscrowAddressRequest>,
+    ) -> std::result::Result<tonic::Response<QueryEscrowAddressResponse>, tonic::Status> {
+        unimplemented!()
```

Comment:
Noting for the future that I think we can implement this by querying the value balance key, even though we don't have escrow accounts per se.